### ISSUE TITLE
`dim_axes` skips dimensions without `axis` attribute

### ISSUE THREAD

**Issue by rod-glover** [pacificclimate/nchelpers#14]
Regarding the code block commented "Then fill in the rest from the 'axis' attributes" in method `dim_axes`, @jameshiebert commented in a recent PR:

> I'm a bit concerned about conflating the conditionals of "variable exists" and "axis attribute exists". I.e. if the variable exists, but doesn't have an axis attribute, then this script will just skip it. However, I would want the script to either error out or raise a really loud warning. We should probably write an issue for this and defer.

**Comment by rod-glover** [comment]
Are there legitimate cases where a NetCDF dimension (pragmatically, a variable returned by the `dimensions` method) is not attributed with `axis`? If so, we have to allow for that.

**Comment by jameshiebert** [comment]
Yep. Happens all the time. The beauty and horror of NetCDF is that you can create a file in the format in almost any manner that you want. That's why you've had to work so hard on the metadata standards... without them you have to expect anything.

So maybe a way of rephrasing your question is: if a file doesn't obey our metadata standards, what do we do? Since our metadata standards are geared towards processing and indexing, I'd argue that we can assume that they exist in the code for those applications (e.g. in ``modelmeta`` and ``climate-explorer-backend``). But this software is intended to be more general, so we kind of have to expect the  worst.

**Comment by rod-glover** [comment]
I suppose we could have a "strict mode" in nchelpers that, if enabled, raises an exception in cases like this.

**Comment by rod-glover** [comment]
Heh, to accommodate a couple of cases, make strict mode it a kw parameter of `CFDataset`, with possible values `None` (non strict mode) or a function that is to be called when violation is detected. Then you get code like:

```
cf = CFDataset('file.nc', strict_mode=logger.error)
cf = CFDataset('file.nc', strict_mode=lambda msg: raise Exception(msg))
```
